gh-132734: Add new constants for Bluetooth sockets (#132735)
python/cpython: Doc/library/socket.rst

   These constants describe the Bluetooth address type when binding or
   connecting a :const:`BTPROTO_L2CAP` socket.

   .. availability:: Linux, FreeBSD

   .. versionadded:: 3.14

.. data:: SOL_RFCOMM
          SOL_L2CAP
          SOL_HCI
          SOL_SCO
          SOL_BLUETOOTH

   Used in the level argument to the :meth:`~socket.setsockopt` and
   :meth:`~socket.getsockopt` methods of Bluetooth socket objects.

>    :const:`SOL_BLUETOOTH` is only available on Linux, other constants

Reviewer: ```suggestion
   :const:`SOL_BLUETOOTH` is only available on Linux. Other constants
```

   are available if the corresponding protocol is supported.

.. data:: SO_L2CAP_*
          L2CAP_LM
          L2CAP_LM_*
          SO_RFCOMM_*
          RFCOMM_LM_*
          SO_SCO_*
          SO_BTH_*
          BT_*

   Used in the option name and value argument to the :meth:`~socket.setsockopt`
   and :meth:`~socket.getsockopt` methods of Bluetooth socket objects.

   :const:`!BT_*` and :const:`L2CAP_LM` are only available on Linux.
   :const:`!SO_BTH_*` are only available on Windows.

>    Other constants may be available on Linux and varios BSD platforms.

Reviewer: ```suggestion
   Other constants may be available on Linux and various BSD platforms.
```

   .. versionadded:: next

.. data:: HCI_FILTER
          HCI_TIME_STAMP
          HCI_DATA_DIR
          SO_HCI_EVT_FILTER
          SO_HCI_PKT_FILTER

>    Options names for use with :const:`BTPROTO_HCI`.
>    Availability and format of the option value depend on platform.

Reviewer: ```suggestion
   Option names for use with :const:`BTPROTO_HCI`.
   Availability and format of the option values depend on platform.
```

   .. availability:: Linux, FreeBSD

   .. versionadded:: 3.14

.. data:: SOL_RFCOMM
          SOL_L2CAP
          SOL_HCI
          SOL_SCO
          SOL_BLUETOOTH

   Values to specify level for Bluetooth socket options in
   :func:`setsockopt` and :func:`getsockopt`.

   .. availability::
      :const:`SOL_BLUETOOTH` is only available on Linux, other constants
      are available if the corresponding protocol is supported.

.. data:: SO_L2CAP_*
          L2CAP_LM
          L2CAP_LM_*
          SO_RFCOMM_*
          RFCOMM_LM_*
          SO_SCO_*
          SO_BTH_*
          BT_*

   Option names and values for Bluetooth socket options in
   :func:`setsockopt` and :func:`getsockopt`.

   .. availability::
      :const:`!BT_*` :const:`L2CAP_LM` are only available on Linux.
      :const:`!SO_BTH_*` are only available on Windows.
      Other constants may be available on Linux and BSD.

   .. versionadded:: next

.. data:: HCI_FILTER
          HCI_TIME_STAMP
          HCI_DATA_DIR
          SO_HCI_EVT_FILTER
          SO_HCI_PKT_FILTER

   Options names for use with :const:`BTPROTO_HCI`.
   Availability and format of the option value depend on platform.

>    .. availability:: Linux and BSD

Reviewer 2: ```suggestion
   .. availability:: Linux, FreeBSD
```

Consistent with your first one, also we cannot officially guarantee *BSD support

Author: It is not only FreeBSD. In several places I use BSD as a shortcut for FreeBSD, NetBSD and DragonFly BSD.

Note that different constants are available on different platforms.

   .. versionchanged:: next
      Added :const:`!SO_HCI_EVT_FILTER` and :const:`!SO_HCI_PKT_FILTER`

>       on NetBSD and DragonFly BSD.
>       Added :const:`!HCI_DATA_DIR` on FreeBSD, NetBSD and DragonFly BSD.

Reviewer 2: ```suggestion
      for NetBSD and DragonFly BSD.
      Added :const:`!HCI_DATA_DIR` for FreeBSD, NetBSD and DragonFly BSD.
```
I think *for* is more fitting here

Author: Is it so? I thought that it is more common to write "on *platform*" than "for *platform*". There are many examples of the former, and not many of the latter.

Reviewer 3: I also prefer "on NetBSD" here.
